cylc/uiserver/resolvers.py

```diff
@@ -534,22 +543,27 @@ async def service(
         workflows: Iterable['Tokens'],
         kwargs: Dict[str, Any]
     ) -> List[Union[bool, str]]:
-        if command == 'clean':
-            return await Services.clean(
+
+        SERVICES = {
+            'clean': Services.clean(
                 workflows,
                 kwargs,
                 self.workflows_mgr,
                 log=self.log,
                 executor=self.executor
-            )
-
-        elif command == 'play':
-            return await Services.play(
+            ),
+            'play': Services.play(
                 workflows,
                 kwargs,
                 self.workflows_mgr,
-                log=self.log
-            )
+                log=self.log,
+            ),
+            'scan': Services.scan(
+                kwargs,
+                self.workflows_mgr,
+            ),
+        }
```

Comment:
We can move this dictionary out of the function to avoid re-declaring it every time the function is run e.g:

```python
SERVICES = { ... }

async def service(...): ...
```

Reply:
This logic has been moved out of the 'service' method and moved into a 'get_service' method.

Initially Flake8 was unhappy with multiple if-else statements and suggested using a dictionary instead, however I struggled to make that work. I also tried using getattr and partial functions to get around the multiple if else - also couldnt make that work. Have now reverted back to the if-else option (which flake8 isn't complaining about now).

getattr and partial are still being used but are not necessary. The get_service method could just be...

`
    async def get_service(self, command, kwargs, workflows):
        if command == 'clean':
            mutation_response = Services.clean(
                workflows,
                kwargs,
                self.workflows_mgr,
                log=self.log,
                executor=self.executor
            )
        elif command == 'play':
            mutation_response = Services.play(
                workflows,
                kwargs,
                self.workflows_mgr,
                log=self.log
            )
        elif command == 'scan':
            mutation_response = Services.scan(
                kwargs,
                self.workflows_mgr
            )
        return await mutation_response()
`